nextflow_schema.json

```diff
@@ -44,43 +47,301 @@
+                    "fa_icon": "fas fa-forward"
+                },
+                "kofam_dir": {
                     "type": "string",
+                    "default": "./kofam",
+                    "description": "Path to a directory with KOfam files. Will be created if it doesn't exist.",
+                    "help_text": "If a `ko_list` file and/or `profiles` does not exist, they will be downloaded.",
+                    "fa_icon": "fas fa-folder-open"
+                },
+                "hmmdir": {
+                    "type": "string",
+                    "description": "Directory with hmm files which will be searched for among ORFs",
+                    "pattern": "^\\S+",
                     "format": "directory-path",
-                    "description": "Directory / URL base for iGenomes references.",
-                    "default": "s3://ngi-igenomes/igenomes",
-                    "fa_icon": "fas fa-cloud-download-alt",
-                    "hidden": true
+                    "fa_icon": "fas fa-folder-open"
                 },
-                "igenomes_ignore": {
+                "hmmfiles": {
+                    "type": "string",
+                    "description": "Comma-separated list of hmm files which will be searched for among ORFs",
+                    "pattern": "\\S+hmm(\\.gz)?",
+                    "format": "file-path",
+                    "mimetype": "text/plain",
+                    "fa_icon": "far fa-file-code"
+                },
+                "hmmpattern": {
+                    "type": "string",
+                    "default": "*.hmm",
+                    "description": "specify which pattern hmm files end with",
+                    "fa_icon": "fas fa-barcode"
+                }
+            },
+            "fa_icon": "fas fa-clipboard"
+        },
+        "taxonomy_annotation_options": {
+            "title": "Taxonomy annotation options",
+            "type": "object",
+            "description": "",
+            "default": "",
+            "properties": {
+                "skip_eukulele": {
                     "type": "boolean",
-                    "description": "Do not load the iGenomes reference config.",
-                    "fa_icon": "fas fa-ban",
-                    "hidden": true,
-                    "help_text": "Do not load `igenomes.config` when running the pipeline. You may choose this option if you observe clashes between custom parameters and those supplied in `igenomes.config`."
+                    "description": "skip eukulele run",
+                    "fa_icon": "fas fa-forward"
+                },
+                "eukulele_method": {
+                    "type": "string",
+                    "default": "mets",
+                    "description": "Specify which method to use for EUKulele. the alternatives are: mets (metatranscriptomics) or  mags (Metagenome Assembled Genomes). default: mets",
+                    "fa_icon": "fas fa-bezier-curve"
+                },
```

Comment:
Use `enum` to specify & restrict to specific option, such as in https://github.com/nf-core/ampliseq/blob/4e48b7100302e2576ac1be2ccc7d464253e9d20e/nextflow_schema.json#L231
might be also interesting for other params